lang/test/org/partiql/lang/eval/evaluatortestframework/AbstractPipeline.kt

```diff
@@ -0,0 +1,15 @@
+package org.partiql.lang.eval.evaluatortestframework
+
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.TypingMode
+
+/**
+ * Represents an abstract pipeline (either [org.partiql.lang.CompilerPipeline] or
+ * [org.partiql.lang.planner.PlannerPipeline]) so that [PipelineEvaluatorTestAdapter] can work with either.
+ *
+ * Includes only those properties and methods that are required for testing purposes.
+ */
+interface AbstractPipeline {
+    val typingMode: TypingMode
```

Comment:
Does it make more sense to pass in `CompileOptions` here? What if we want to evaluate and test the other pipelines with other compiler options in the future? E.g. passing in `typedOpBehavior` or `defaultTimezoneOffset`?

Reply:
The plan evaluator (introduced in #592) has `EvaluatorOptions` instead of `CompileOptions`.  The only thing is truly needed is the `typingMode`, which is common to both of those classes, thus, this was added.